functions.php

```diff
@@ -476,3 +476,55 @@ function ( $downsize, $id, $size ) {
 	100,
 	3
 );
+
+// This action overrides the WordPress functionality for adding a notice message
+// https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-admin/edit-form-blocks.php#L303-L305
+// When it's a page for posts.
+add_action(
+	'admin_enqueue_scripts',
+	function() {
+		global $post;
+
+		if ( $post && (int) get_option( 'page_for_posts' ) === $post->ID ) {
+			// Adding this style, it works as a workaround for editors
+			// To disable the ability to edit the content of the listing page.
+			echo '<style>
+				.edit-post-header-toolbar,
+				.block-list-appender {
+					pointer-events: none;
+					opacity: 0;
+				}
+
+				.block-editor-block-list__layout {
+					display: none;
+				}
+
+				.components-notice__actions {
+					display: inline-flex !important;
+					margin-left: 5px;
+				}
+			</style>';
+
+			wp_add_inline_script(
+				'wp-notices',
```

Comment:
Here is the notices "how to" guideline.
https://developer.wordpress.org/block-editor/how-to-guides/notices/